swift/apple/FirezoneNetworkExtension/Adapter.swift

```diff
@@ -214,17 +192,17 @@ class Adapter {
     networkMonitor?.cancel()
 
     // Shutdown the tunnel
-    if case .tunnelStarted(let session) = self.state {
-      Log.log("Adapter.deinit: Shutting down connlib")
-      session.disconnect()
-    }
+    session?.disconnect()
+
+    session = nil
```

Comment:
I believe this is the important fix. When we call `onDisconnect`, the pointer to the session is still valid. It is when you call `disconnect` that we free the memory.

Arguably, this is a shortcoming of `swift-bridge` I think where it doesn't model correctly, that we capture `self` entirely in the `disconnect` call.

Reply:
Should have RTFM: https://chinedufn.github.io/swift-bridge/safety/index.html#never-use-a-value-after-it-is-dropped